### PR TITLE
 [FLINK-13399][table] Create two separate table uber jars for old and blink planners

### PR DESCRIPTION
**Pull request by dawidwys**
## What is the purpose of the change

This PRs creates two table api uber jars bundling two different planners (the current one and the blink planner). Also for a user convenience both are placed in the lib directory. The current planner should be enabled by default (#9249)

## Verifying this change

The `test_streaming_sql.sh` and `test_sql_client.sh` should run (with the legacy planner)

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (**yes** / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (**yes** / no)
  - If yes, how is the feature documented? (not applicable / **docs** / JavaDocs / not documented)

### CODE REVIEW

File: docs/dev/table/index.md [hunk 1]
```diff
@@ -44,33 +44,48 @@ The following dependencies are relevant for most projects:
 * `flink-table-api-java-bridge`: The Table & SQL API with DataStream/DataSet API support using the Java programming language.
 * `flink-table-api-scala-bridge`: The Table & SQL API with DataStream/DataSet API support using the Scala programming language.
 * `flink-table-planner`: The table program planner and runtime.
```
Comment (twalthr) [inline docs/dev/table/index.md]: Update that this is the old planner of before Flink 1.9 and still the recommended one?

File: docs/dev/table/index.md [hunk 2]
```diff
@@ -44,33 +44,48 @@ The following dependencies are relevant for most projects:
...
-* `flink-table-uber`: Packages the modules above into a distribution for most Table & SQL API use cases. The uber JAR file `flink-table*.jar` is located in the `/opt` directory of a Flink release and can be moved to `/lib` if desired.
+* `flink-table-planner-blink`: The table program blink planner.
+* `flink-table-runtime-blink`: The table program blink runtime.
+* `flink-table-uber`: Packages the common modules above plus the current planner into a distribution for most Table & SQL API use cases. The uber JAR file `flink-table*.jar` is by default located in the `/lib` directory of a Flink release.
```
Comment (twalthr) [inline docs/dev/table/index.md]: What does "common modules above" mean? Maybe "Packages the API plus the old planner and runtime?"

Comment (twalthr) [inline docs/dev/table/index.md]: `flink-table*.jar` -> `flink-table-uber-VERSION.jar`

File: docs/dev/table/index.md [hunk 3]
```diff
@@ -44,33 +44,48 @@ The following dependencies are relevant for most projects:
...
   <version>{{site.version}}</version>
 </dependency>
 {% endhighlight %}
 
-Additionally, depending on the target programming language, you need to add the Java or Scala API.
-
+Additionally, if you want to run the Table API & SQL programs locally you must add one of the
+following set of modules, depending which planner you want to use:
 {% highlight xml %}
 <!-- Either... -->
```
Comment (twalthr) [inline docs/dev/table/index.md]: "for the old Flink planner before Flink 1.9"

File: docs/dev/table/index.md [hunk 4]
```diff
@@ -44,33 +44,48 @@ The following dependencies are relevant for most projects:
 * `flink-table-api-java-bridge`: The Table & SQL API with DataStream/DataSet API support using the Java programming language.
```
Comment (twalthr) [inline docs/dev/table/index.md]: Maybe some introductory words at the very beginning that we have two planners?

File: flink-table/flink-table-runtime-blink/pom.xml [hunk 5]
```diff
@@ -170,6 +170,12 @@ under the License.
 									<include>org.lz4:lz4-java</include>
 								</includes>
 							</artifactSet>
+							<relocations>
+								<relocation>
+									<pattern>net.jpountz</pattern>
```
Comment (twalthr) [inline flink-table/flink-table-runtime-blink/pom.xml]: where is this coming from?

File: flink-table/flink-table-uber-blink/pom.xml [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+		<artifactId>flink-table</artifactId>
+		<version>1.10-SNAPSHOT</version>
+		<relativePath>..</relativePath>
+	</parent>
+
+	<artifactId>flink-table-uber-blink_${scala.binary.version}</artifactId>
+	<name>flink-table-uber-blink</name>
+	<description>
+		This module contains the entire Table/SQL distribution for writing table programs
+		within the table ecosystem or between other Flink APIs. This module uses the blink planner.
```
Comment (twalthr) [inline flink-table/flink-table-uber-blink/pom.xml]: nit: `Blink planner`
Maybe explain what the planner is in a sentence.

**Comment by wuchong** [comment]
Sure @twalthr , @docete  will look into the streaming e2e test. We also find the streaming e2e is not stable on travis. We will look into both problems.

File: docs/dev/table/index.md (hunk 2 above)
Comment (dawidwys) [reply docs/dev/table/index.md]: I renamed it delta the fact that when copying the uber jar we are stripping the `uber` suffix. This is the current behavior, which I adopted for the other uber jar as well.

So I renamed it to `flink-table-VERSION.jar`

File: flink-table/flink-table-runtime-blink/pom.xml (hunk 5 above)
Comment (dawidwys) [reply flink-table/flink-table-runtime-blink/pom.xml]: This the package of `org.lz4:lz4-java`

**Comment by docete** [comment]
> Sure @twalthr , @docete will look into the streaming e2e test. We also find the streaming e2e is not stable on travis (FLINK-13377). We will look into both problems.

@twalthr @wuchong I verified stream sql e2e test offline and found [FLINK-13494](https://issues.apache.org/jira/browse/FLINK-13494) caused fails for blink planner. @XuPingyong will open a PR for FLINK-13494 and I will open a PR for FLINK-13439 soon.

**Comment by twalthr** [comment]
Thanks for working on this @docete. I will have a final pass through this PR and merge it, if there are no objections? @dawidwys

**Pull request closed without merging**